thunx-encoding-json/src/main/java/com/contentgrid/thunx/encoding/json/JsonCollectionValueDto.java

```diff
@@ -0,0 +1,78 @@
+package com.contentgrid.thunx.encoding.json;
+
+import com.contentgrid.thunx.predicates.model.CollectionValue;
+import com.contentgrid.thunx.predicates.model.Scalar;
+import com.contentgrid.thunx.predicates.model.ThunkExpression;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Data
+@EqualsAndHashCode(callSuper=true)
+@NoArgsConstructor
+public class JsonCollectionValueDto extends JsonScalarDto<Collection<JsonScalarDto<?>>> {
```

Comment:
I'm not sure that this should extend `JsonScalarDto`, as it's not really a scalar value.

Additionally, since you have basically 2 completely separate implementations, depending on whether it is a set or an array; should'nt this just be 2 completely different classes

> A scalar data type, or just scalar, is any non-composite value.
> -- https://en.wikipedia.org/wiki/Scalar_processor#Scalar_data_type